### PR TITLE
fix cors problems with newsfeed articles

### PR DESCRIPTION
**Pull request by khassel**
solves #2840 as far as possible. There could still be errors on the embedded iframe when the owner of the site has set `X-Frame-Options` or `Access-Control-Allow-Origin` headers (as already mentioned in the docs).

### CODE REVIEW

**Comment by rejas** [comment]
Seems to look good, but I'd like to wait for my approval until Tuesday when I return from my trip

File: modules/default/newsfeed/newsfeed.js
Comment (rejas) [inline modules/default/newsfeed/newsfeed.js:45]: so this will always be used even if cors is not necessary?

Comment (khassel) [reply modules/default/newsfeed/newsfeed.js:45]: yes, I thought about a config option but as it works so too I decided against.

Comment (rejas) [reply modules/default/newsfeed/newsfeed.js:45]: Does that have any downsides? I dont have much knowledge into cors, so I am afraif sites might not be loading if this cors hack is used all of the time?

Comment (khassel) [reply modules/default/newsfeed/newsfeed.js:45]: I found no downsides.

I tested with the urls provided in #2840 and with mine, no one did work without the cors prefix. These are not the newsfeed url's but the url's of the articles and they are only used if you want read the full article (which I think is not very often).

Comment (rejas) [reply modules/default/newsfeed/newsfeed.js:45]: I would feel better if we had a config option to disable cors so that we could point users that run into an issue to it when the next release is out.

Comment (khassel) [reply modules/default/newsfeed/newsfeed.js:45]: so on what level do you see this config option? One option for whole newsfeed module or option for every feed?

And which should be the default? From my side I would enable the cors proxy by default.

Comment (rejas) [reply modules/default/newsfeed/newsfeed.js:45]: Yes, I would enable the cors proxy by default too. 
And have the config option per newsfeed, not for the whole module.

File: modules/default/newsfeed/newsfeedfetcher.js
Comment (rejas) [inline modules/default/newsfeed/newsfeedfetcher.js:21]: Missing JSDoc @param "useCorsProxy" declaration

**Review by rejas — CHANGES_REQUESTED** [review]
just one small jsdoc addition needed :-)

File: modules/default/newsfeed/newsfeedfetcher.js
Comment (khassel) [reply modules/default/newsfeed/newsfeedfetcher.js:22]: fixed ... but should these warning not treated as errors? Or will you check every test manually in the future?

Comment (rejas) [reply modules/default/newsfeed/newsfeedfetcher.js:22]: na, wont check manually since the warning shows up here in the changes ... erroring it might trigger the old ones, I fear...

**Review by rejas — APPROVED** [review]

**Pull request merged**
